projects/plugins/jetpack/_inc/client/state/site/reducer.js

```diff
@@ -45,6 +48,8 @@ export const data = ( state = {}, action ) => {
 			return assign( {}, state, action.siteData );
 		case JETPACK_SITE_BENEFITS_FETCH_RECEIVE:
 			return merge( {}, state, { site: { benefits: action.siteBenefits } } );
+		case JETPACK_SITE_DISCOUNT_FETCH_RECEIVE:
+			return merge( {}, state, { site: { discount: action.siteDiscount } } );
```

Comment:
I tested at one point with an already connected site and noticed the request to get the coupon can come back with an empty code. Do we still want to store it in state in that case?

Reply:
Good call.